src/main/java/org/folio/service/transactions/BatchTransactionService.java

```diff
@@ -54,25 +94,40 @@ private void checkIdIsPresent(List<Transaction> transactions, String operation)
       if (StringUtils.isBlank(transaction.getId())) {
         throw new HttpException(400, String.format("Id is required in transactions to %s.", operation));
       }
+      if (TYPES_WITH_SUMMARIES.contains(transaction.getTransactionType()) && transaction.getSourceInvoiceId() == null &&
+          (transaction.getEncumbrance() == null || transaction.getEncumbrance().getSourcePurchaseOrderId() == null)) {
```

Comment:
ids better to check with StringUtils to avoid empty strings

Reply:
OK for isBlank.
For some transaction types, is an error if there is no order id and no invoice id. The test simply checks that. Payment and Credit don't have an order id but they need to have an invoice id.